frontend/web/src/assets/lib/jsc3d/jsc3d.webgl.js

```diff
@@ -0,0 +1,1202 @@
+
+	gl.useProgram(this.programs.picking);
+
+	for(var i=0; i<renderList.length; i++) {
+		var mesh = renderList[i];
+		if(mesh.isTrivial() || !mesh.visible)
+			continue;
+
+		// skip the mesh if it is nearly completely transparent
+		var material = mesh.material || defaultMaterial;
+		if(material.transparency > 0.99)
+			continue;
+
+		if(isCullingDisabled || mesh.isDoubleSided)
+			gl.disable(gl.CULL_FACE);
+		else
+			gl.enable(gl.CULL_FACE);
+
+		gl.uniformMatrix4fv(this.programs.picking.uniforms['u_transformMatrix'], false, transformMat4);
+		gl.uniform3fv(this.programs.picking.uniforms['u_pickingId'], mesh.compiled.pickingId);
+		gl.enableVertexAttribArray(this.programs.picking.attributes['a_position']);
+
+		switch(mesh.compiled.remderMode) {
+		case 'point':
+			gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.coords);
+			gl.vertexAttribPointer(this.programs.picking.attributes['a_position'], 3, gl.FLOAT, false, 0, 0);
+			gl.drawArrays(gl.POINTS, 0, mesh.compiled.coordCount);
+			break;
+		case 'wireframe':
+			gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.edges);
+			gl.vertexAttribPointer(this.programs.picking.attributes['a_position'], 3, gl.FLOAT, false, 0, 0);
+			gl.drawArrays(gl.LINES, 0, mesh.compiled.edgeCount);
+			break;
+		case 'flat':
+		case 'smooth':
+		case 'texture':
+		case 'textureflat':
+		case 'texturesmooth':
+		default:
+			gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.coords);
+			gl.vertexAttribPointer(this.programs.picking.attributes['a_position'], 3, gl.FLOAT, false, 0, 0);
+			gl.drawArrays(gl.TRIANGLES, 0, mesh.compiled.coordCount);
+			break;
+		}
+	}
+};
+
+/**
+ * Compile a mesh according to the given render mode, generating the WebGL dependent stuff.
+ * @private
+ */
+JSC3D.WebGLRenderBackend.prototype.compileMesh = function(mesh, renderMode) {
+	if(mesh.isTrivial())
+		return false;
+
+	renderMode = mesh.renderMode || renderMode;
+
+	function makeWireframe(ibuf, vbuf, numOfFaces, trianglesOnly) {
+		var edges;
+
+		var v0, v1, v2;
+		if(trianglesOnly) {
+			edges = new Float32Array(18 * numOfFaces);
+			for(var i=0, e=0; i<ibuf.length; i+=4, e+=18) {
+				v0 = 3 * ibuf[i    ];
+				v1 = 3 * ibuf[i + 1];
+				v2 = 3 * ibuf[i + 2];
+
+				// v0 <-> v1
+				edges[e     ] = vbuf[v0    ];
+				edges[e +  1] = vbuf[v0 + 1];
+				edges[e +  2] = vbuf[v0 + 2];
+				edges[e +  3] = vbuf[v1    ];
+				edges[e +  4] = vbuf[v1 + 1];
+				edges[e +  5] = vbuf[v1 + 2];
+				// v1 <-> v2
+				edges[e +  6] = vbuf[v1    ];
+				edges[e +  7] = vbuf[v1 + 1];
+				edges[e +  8] = vbuf[v1 + 2];
+				edges[e +  9] = vbuf[v2    ];
+				edges[e + 10] = vbuf[v2 + 1];
+				edges[e + 11] = vbuf[v2 + 2];
+				// v2 <-> v0
+				edges[e + 12] = vbuf[v2    ];
+				edges[e + 13] = vbuf[v2 + 1];
+				edges[e + 14] = vbuf[v2 + 2];
+				edges[e + 15] = vbuf[v0    ];
+				edges[e + 16] = vbuf[v0 + 1];
+				edges[e + 17] = vbuf[v0 + 2];
+			}
+		}
+		else {
+			edges = [];
+			for(var i=0, j=0; i<numOfFaces; i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "i" as this name is already used in declaration at line 841. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)